hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/web/resources/NamenodeWebHdfsMethods.java

```diff
@@ -311,25 +315,15 @@ static DatanodeInfo chooseDatanode(final NameNode namenode,
       }
     }
 
-    if (op == PutOpParam.Op.CREATE) {
```

Comment:
This is the actual change. It used to treat `CREATE` special by calling the BlockManager.
This removes the special case and instead makes calling the BlockManager the norm for all requests except OPEN, APPEND and GETFILECHECKSUM for which it will pick an actual replica.